.github/workflows/update_notebooks.yml

```diff
@@ -0,0 +1,78 @@
+name: Run docs notebooks, check in results
+
+on:
+  workflow_dispatch: # This allows the workflow to be triggered manually
+
+env:
+  POETRY_VERSION: "1.7.1"
+
+jobs:
+  update-and-pr:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Set up Git
+      run: |
+        git config --global user.name 'github-actions[bot]'
+        git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: "./langchain/.github/actions/poetry_setup"
+      with:
+        python-version: ${{ matrix.python-version }}
+        poetry-version: ${{ env.POETRY_VERSION }}
+        working-directory: langchain/${{ matrix.working-directory }}
+        cache-key: scheduled
+
+    - name: Install dependencies
+      run: |
+        poetry run pip install -U pip
+        poetry run pip install -U jupyter nbconvert
+
+    - name: Run how-to notebooks
+      env:
+        OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+        ANTHROPIC_API_KEY: ${{ secrets.ANTHROPIC_API_KEY }}
+        AZURE_OPENAI_API_VERSION: ${{ secrets.AZURE_OPENAI_API_VERSION }}
+        AZURE_OPENAI_API_BASE: ${{ secrets.AZURE_OPENAI_API_BASE }}
+        AZURE_OPENAI_API_KEY: ${{ secrets.AZURE_OPENAI_API_KEY }}
+        AZURE_OPENAI_CHAT_DEPLOYMENT_NAME: ${{ secrets.AZURE_OPENAI_CHAT_DEPLOYMENT_NAME }}
+        AZURE_OPENAI_LLM_DEPLOYMENT_NAME: ${{ secrets.AZURE_OPENAI_LLM_DEPLOYMENT_NAME }}
+        AZURE_OPENAI_EMBEDDINGS_DEPLOYMENT_NAME: ${{ secrets.AZURE_OPENAI_EMBEDDINGS_DEPLOYMENT_NAME }}
+        FIREWORKS_API_KEY: ${{ secrets.FIREWORKS_API_KEY }}
+        GROQ_API_KEY: ${{ secrets.GROQ_API_KEY }}
+        HUGGINGFACEHUB_API_TOKEN: ${{ secrets.HUGGINGFACEHUB_API_TOKEN }}
+        MISTRAL_API_KEY: ${{ secrets.MISTRAL_API_KEY }}
+        COHERE_API_KEY: ${{ secrets.COHERE_API_KEY }}
+        NVIDIA_API_KEY: ${{ secrets.NVIDIA_API_KEY }}
+        GOOGLE_API_KEY: ${{ secrets.GOOGLE_API_KEY }}
+        GOOGLE_SEARCH_API_KEY: ${{ secrets.GOOGLE_SEARCH_API_KEY }}
+        GOOGLE_CSE_ID: ${{ secrets.GOOGLE_CSE_ID }}
+        # TODO: Remove this once 0.3 is relased
+        PIP_PRE: 1
+      run: |
+        poetry run jupyter nbconvert --to notebook --execute --allow-errors docs/docs/how_to/*.ipynb;
```

Comment:
Could you document what this does?

Why are we allowing for errors?

Reply:
If I had to guess, this is temporary until we fix em

If not, we should make this temporary until we fix the notebooks intentional error cells to mark them as "error allowed" (I think jupyter has a concept of this)